extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -69,14 +69,26 @@ export async function launchPublishDatabaseQuickpick(project: Project): Promise<
 		// so exit the flow.
 		return;
 	}
+	quickPick.hide(); // Hide the quickpick immediately so it isn't showing while the API loads
 
 	// 2. Select connection
-	const api = await getVscodeMssqlApi();
-	const connectionProfile = await api.promptForConnection();
-	if (!connectionProfile) {
-		return;
+	const vscodeMssqlApi = await getVscodeMssqlApi();
+	let dbs: string[] | undefined = undefined;
+	while (!dbs) {
+		const connectionProfile = await vscodeMssqlApi.promptForConnection(true);
+		if (!connectionProfile) {
+			return;
+		}
+		// Get the list of databases now to validate that the connection is valid and re-prompt them if it is
```

Comment:
are you missing a word 'not' at the end?

Reply:
I prefer to leave people reading my code in suspense 😉